providers/pool.rb

```diff
@@ -120,120 +120,152 @@ def site_identifier
-      cmd << " /+recycling.periodicRestart.schedule.[value='#{@new_resource.recycle_at_time}']"
+      cmd << " /autoStart:#{@new_resource.auto_start.to_s}" if is_new_auto_start
+      cmd << " /startMode:#{@new_resource.start_mode.to_s}" if is_new_start_mode
+      cmd << " /managedRuntimeVersion:v#{@new_resource.runtime_version}" if @new_resource.runtime_version && is_new_managed_runtime_version
+      cmd << " /recycling.periodicRestart.time:#{@new_resource.recycle_after_time}" if @new_resource.recycle_after_time && is_new_recycle_after_time
+      cmd << " /enable32BitAppOnWin64:#{@new_resource.thirty_two_bit}" if @new_resource.thirty_two_bit && is_new_enable_32_bit_app_on_win_64
+      cmd << " /-recycling.periodicRestart.schedule" if @new_resource.recycle_at_time && is_new_recycle_at_time
+      cmd << " /+recycling.periodicRestart.schedule.[value='#{@new_resource.recycle_at_time}']" if @new_resource.recycle_at_time && is_new_recycle_at_time
+      cmd << " /processModel.maxProcesses:#{@new_resource.max_proc}" if @new_resource.max_proc && is_new_max_processes
+      cmd << " /queueLength:#{@new_resource.queue_length}" if @new_resource.queue_length && is_new_queue_length
       Chef::Log.debug(cmd)
       shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.runtime_version && is_new_managed_runtime_version
-      was_updated = true
-      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
-      cmd << " /managedRuntimeVersion:v#{@new_resource.runtime_version}"
-      Chef::Log.debug(cmd) if @new_resource.runtime_version
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.worker_idle_timeout && is_new_idle_timeout
-      was_updated = true
-      cmd = "#{appcmd(node)} set config /section:applicationPools"
-      cmd << " \"/[name='#{@new_resource.pool_name}'].processModel.idleTimeout:#{@new_resource.worker_idle_timeout}\""
-      Chef::Log.debug(cmd)
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
     end
+
+    # Application Pool Config
+    # processModel items
+    configure_application_pool(@new_resource.worker_idle_timeout && is_new_idle_timeout, "processModel.idleTimeout:#{@new_resource.worker_idle_timeout}")
+    configure_application_pool(is_new_load_user_profile, "processModel.loadUserProfile:#{@new_resource.load_user_profile}")
+    configure_application_pool(is_new_set_profile_environment, "processModel.setProfileEnvironment:#{@new_resource.set_profile_environment}")
+    configure_application_pool(is_new_logon_type, "processModel.logonType:#{@new_resource.logon_type}")
+    configure_application_pool(is_new_manual_group_membership, "processModel.manualGroupMembership:#{@new_resource.manual_group_membership}")
+    configure_application_pool(is_new_idle_timeout, "processModel.idleTimeout:#{@new_resource.idle_timeout}")
```

Comment:
May be nothing, but should this line (214) be using the same is_new_idle_timeout as linke (209) above.  Seems like this line would always override the previous..?